play-json-ops-common/src/main/scala/play/api/libs/json/ops/FormatOps.scala

```diff
@@ -8,6 +8,24 @@ import scala.util.control.NonFatal
 
 object FormatOps {
 
+  /**
+   * Extracts the class name of the [[Enumeration]] removing any `$` symbols without throwing an exception.
+   *
+   * @note this will include any outer object or class names separated by `.`s
+   */
+  def enumClassName(o: Enumeration): String = {
+    // This logic is designed to be robust without much noise
+    // 1. use getName to avoid runtime exceptions from getSimpleName
+    // 2. filter out '$' anonymous class / method separators
+    // 3. start the full class name from the first upper-cased outer class name
+    //    (to avoid picking up unnecessary package names)
+    o.getClass.getName
+      .split('.')
+      .last
+      .split('$')
+      .mkString(".")
```

Comment:
this implementation is the simplest and surprisingly also the fastest out of a few alternatives that I tried:

```scala
// Mean duration per invocation over 50,000 iterations
// classNameNaive: 5.074039199999959E-4 ms
// classNameFromView: 0.0013898413000000196 ms
// classNameFromStringBuilder: 0.001559349300000045

  def classNameNaive(o: AnyRef): String = {
    o.getClass.getName
      .split('.')
      .last
      .split('$')
      .mkString(".")
  }

  def classNameFromView(o: AnyRef): String = {
    o.getClass.getName
      .reverseIterator
      .takeWhile(_ != '.')
      .mkString
      .split('$')
      .reverseIterator
      .filterNot(_.isEmpty)
      .dropWhile(_.head.isLower)
      .mkString(".")
  }

  def classNameFromStringBuilder(o: AnyRef): String = {
    val sb = new StringBuilder
    for (c <- o.getClass.getName.reverseIterator.takeWhile(_ != '.')) {
      sb.append(if (c == '$') '.' else c)
    }
    sb.dropWhile(_ == '$')
    sb.reverseInPlace()
    sb.dropWhile(_ == '$')
    sb.result()
  }
```

Reply:
Maybe something like,
```scala
    o.getClass.getName
      .split('.')
      .lastOption.map(_.split('$').mkString("."))
      .getOrElse(o.getClass.getName)
```